mathext/hyp2f1.go

```diff
@@ -0,0 +1,466 @@
+package mathext
```

Comment:
We will need license notes for all the files.

Reply:
License notes have been added to all newly introduced files.